warehouse/query-core/src/test/java/datawave/query/jexl/visitors/QueryPruningVisitorTest.java

```diff
@@ -46,18 +49,48 @@ public void delayedGeowaveTest() throws ParseException {
         
         JexlNode rebuilt = QueryPruningVisitor.reduce(script, false);
         
-        Assert.assertEquals(
-                        "geowave:intersects(GEO_FIELD, 'POINT(0 0)') && ((_Delayed_ = true) && (GEO_FIELD == '00' || GEO_FIELD == '0100' || GEO_FIELD == '0103'))",
+        assertEquals("geowave:intersects(GEO_FIELD, 'POINT(0 0)') && ((_Delayed_ = true) && (GEO_FIELD == '00' || GEO_FIELD == '0100' || GEO_FIELD == '0103'))",
```

Comment:
If I update the test method to call flattenAll before comparing the actual and expected query strings, all of your new tests pass, except `testGeoQueryProblem`, but that test fails because flattenAll removed _more_ reference expressions than your test expected (which seems desireable based on the changes you're making here).